### PR TITLE
Dedup the logic of plugin validation for a specific Profile

### PR DESCRIPTION
**Pull request by chendave**
Signed-off-by: Dave Chen <dave.chen@arm.com>

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/profile/profile.go [hunk 1]
```diff
@@ -99,6 +99,9 @@ func (v *cfgValidator) validate(cfg config.KubeSchedulerProfile) error {
 	if len(cfg.Plugins.QueueSort.Enabled) != 1 {
```
Comment (damemi) [inline pkg/scheduler/profile/profile.go:99]: the old check would return separate errors for `len == 0` and `len > 1`, could we copy that here?

**Comment by alculquicondor** [comment]
/close

both checks are necessary:

- There are users of the runtime that are not users of profiles (cluster autoscaler)
- The profile validator does checks across multiple profiles, and for that it needs to check that there is one queue sort plugin that matches.

**Comment by chendave** [comment]
Did a search and do find it is used here in the autoscaler,  thanks for the info!


```golang
//cluster-autoscaler/simulator/scheduler_based_predicates_checker.go
	framework, err := schedulerframeworkruntime.NewFramework(
		scheduler_plugins.NewInTreeRegistry(),
		kubeSchedulerProfile,
		schedulerframeworkruntime.WithInformerFactory(informerFactory),
		schedulerframeworkruntime.WithSnapshotSharedLister(sharedLister),
	)
```

**Pull request closed without merging**